File: node-api/get-node-metrics/request.cs
using System;
using System.Threading.Tasks;
using NetCasperSDK;
using NetCasperSDK.JsonRpc;

namespace CasperIntegrations
{
    public class GetNodeMetrics
    {
        public static async Task Main(string[] args)
        {
            string nodeAddress = "http://3.136.227.9:8888/metrics";

            try
            {
                var metrics = await NetCasperClient.GetNodeMetrics(nodeAddress);
                Console.WriteLine(metrics);
            }
            catch (RpcClientException e)
            {
                Console.WriteLine("ERROR:\n" + e.RpcError.Message);
            }
            catch (Exception e)
            {
                Console.WriteLine(e);
            }
        }
    }
}

File: node-api/get-node-metrics/request.py
import json
import os

import pycspr



# A known casper test-net node address.
_NODE_ADDRESS: str = os.getenv("CASPER_NODE_ADDRESS", "3.136.227.9")


def main():
    """Retrieves node metrics.
    
    """
    # Set client.
    client = pycspr.NodeClient(pycspr.NodeConnection(host=_NODE_ADDRESS))

    # Get node metrics.
    metrics: list = client.get_node_metrics()
    
    print("-----------------------------------------------------------------------------------------------------")
    print(f"QUERIED TEST-NET NODE {_NODE_ADDRESS}")
    print("-----------------------------------------------------------------------------------------------------")
    for line in metrics:
        print(f"{line}")
    print("-----------------------------------------------------------------------------------------------------")


if __name__ == "__main__":
    try:
        main()
    except Exception as err:
        print(f"API ERROR @ NODE {_NODE_ADDRESS} :: {err}")
